cartographer/mapping_2d/global_trajectory_builder.cc

```diff
@@ -35,6 +36,17 @@ const Submaps* GlobalTrajectoryBuilder::submaps() const {
 void GlobalTrajectoryBuilder::AddRangefinderData(
     const common::Time time, const Eigen::Vector3f& origin,
     const sensor::PointCloud& ranges) {
+  // TODO average laser scans instead of discarding them?
```

Comment:
I agree, this is indeed a stop gap solution, I was not sure if I should submit a pull request for this at all. I decided to go for it because it really made a big improvement on my bag from #193. 

I will take a look at the 3D case and try to do it that way. If it also gives good results, I will update the pull request and replace scan dropping with scan accumulation.

The TODO was just an idea I thought would make sense - by "averaging" I meant averaging the rangefinder data. E.g. the distance in some direction theta is the average of the distances in the same direction in these sequential scans which would otherwise be dropped. Perhaps it's the same thing as the scan accumulation from 3D that you have mentioned - at this point I can't say because I haven't looked into it yet.